jobs/build.Jenkinsfile

```diff
@@ -458,20 +458,13 @@ lock(resource: "build-${params.STREAM}") {
             // our pipeline environment.
             //
             // First, define a list of all the derivative artifacts for this architecture.
-            def artifacts = ["Aliyun", "AWS", "Azure", "AzureStack", "DigitalOcean", "Exoscale",
-                             "GCP", "IBMCloud", "Nutanix", "OpenStack", "VirtualBox", "VMware", "Vultr"]
-            // For all architectures we need to build the metal/metal4k artifacts and the Live ISO. Since the
-            // ISO depends on the Metal/Metal4k images we'll make sure to put the Metal* ones in the first run
-            // and the Live ISO in the second run.
-            artifacts.add(0, "Metal")
-            artifacts.add(1, "Metal4k")
-            artifacts.add("Live")
+            def artifacts = pipeutils.get_artifacts(pipecfg, params.STREAM, basearch)
             // Run the two runs of parallel builds
             parallelruns = artifacts.collectEntries {
                 [it, {
                     def cmd = it.toLowerCase()
                     shwrap("""
-                    cosa buildextend-${cmd}
+                    cosa buildextend-"${cmd}"
```

Comment:
Do we actually need the quotes here?

Reply:
I wouldn't think so. If we do I'd put it around the entire arg: `"buildextend-${cmd}"

Reply:
I know what happened. 
I was getting the following error, when I changed it for testing, but the really issue is: I skipped  `buildextend-qemu`. Seems it should have a better error about the missing qemu or it could be even a bug?
```
21:32:48  + set -xeuo pipefail
21:32:48  ++ umask
21:32:48  + '[' 0022 = 0000 ']'
21:32:48  + cosa buildextend-aws
21:32:48  [Pipeline] sh
21:32:48  2022-09-29 00:32:48,709 INFO - symlink is for a build and publish command
21:32:48  2022-09-29 00:32:48,709 INFO - ore target aws found via symlink
21:32:48  2022-09-29 00:32:48,711 INFO - Targeting architecture: x86_64
21:32:48  2022-09-29 00:32:48,711 INFO - Targeting build: 412.86.202209290026-0
21:32:48  2022-09-29 00:32:48,715 INFO - Processed build for: OpenShift 4 (RHCOS-x86_64) 412.86.202209290026-0
21:32:48  2022-09-29 00:32:48,715 INFO - operating on rhcos-412.86.202209290026-0-aws.x86_64.vmdk
21:32:48  2022-09-29 00:32:48,715 INFO - Processing the build artifacts
21:32:48  2022-09-29 00:32:48,739 INFO - Staging temp image: /home/jenkins/agent/workspace/build-artifacts-Renata/tmp/tmpv_edgaee/rhcos-412.86.202209290026-0-aws.x86_64.vmdk
21:32:48  Traceback (most recent call last):
21:32:48    File "/usr/lib/coreos-assembler/cmd-buildextend-aws", line 115, in <module>
21:32:48      build.ensure_built()
21:32:48    File "/usr/lib/coreos-assembler/cosalib/build.py", line 263, in ensure_built
21:32:48      self.build_artifacts()
21:32:48    File "/usr/lib/coreos-assembler/cosalib/build.py", line 414, in build_artifacts
21:32:48      self._build_artifacts(*args, **kwargs)
21:32:48    File "/usr/lib/coreos-assembler/cosalib/qemuvariants.py", line 335, in _build_artifacts
21:32:48      meta_patch = self.mutate_image()
21:32:48    File "/usr/lib/coreos-assembler/cosalib/qemuvariants.py", line 251, in mutate_image
21:32:48      self.set_platform()
21:32:48    File "/usr/lib/coreos-assembler/cosalib/qemuvariants.py", line 226, in set_platform
21:32:48      self.image_qemu, self.tmp_image, self.platform])
21:32:48    File "/usr/lib/coreos-assembler/cosalib/qemuvariants.py", line 198, in image_qemu
21:32:48      qimage = os.path.join(
21:32:48    File "/usr/lib64/python3.10/posixpath.py", line 90, in join
21:32:48      genericpath._check_arg_types('join', a, *p)
21:32:48    File "/usr/lib64/python3.10/genericpath.py", line 152, in _check_arg_types
21:32:48      raise TypeError(f'{funcname}() argument must be str, bytes, or '
21:32:48  TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
21:32:48  error: failed to execute cmd-buildextend-aws: exit status 1
```